### DISCUSSION TITLE
Microsoft.AspNetCore dependency missing ?

### DISCUSSION THREAD

**Discussion by Bartmax** [Dotnet-Boxed/Templates#171] [Q&A]
it didn't run until I added this to csproj.
```
<PackageReference Include="Microsoft.AspNetCore" Version="1.1.2" />
```

**Comment by RehanSaeed** [comment]
Thats a meta package that contains references to a bunch of default packages. See https://www.nuget.org/packages/Microsoft.AspNetCore. Which template did you create?

**Comment by Bartmax** [comment]
this one: https://github.com/ASP-NET-Core-Boilerplate/Templates/blob/master/MVC%206%20API.md

**Comment by RehanSaeed** [comment]
What error were you getting before adding that package?

**Comment by Bartmax** [comment]
``` 
Error	CS1061	
'WebHostBuilder' does not contain a definition for 'UseKestrel' 
and no extension method 'UseKestrel' accepting a first argument of type 'WebHostBuilder' 
could be found (are you missing a using directive or an assembly reference?)

``` 

![image](https://user-images.githubusercontent.com/3386797/27182606-300b5442-51b2-11e7-99e5-d39ec1029fde.png)

**Comment by RehanSaeed** [comment]
That method lives in the `Microsoft.AspNetCore.Server.Kestrel` package which is only removed if you select `WebListener` for the `PrimaryWebServer` setting (`Kestrel` is the default). Did you select `WebListener` and then try to add `Kestrel` manually? The position of the `UseKestrel` line is not what the template outputs.

**Comment by Bartmax** [comment]
at this point im not sure, I will close and reopen if I could repro with more detailed and concrete info, as it is right now is not very actionable. It may be indeed a problem on my end
